Lib/ast.py

```diff
@@ -305,28 +306,13 @@ def get_docstring(node, clean=True):
     return text
 
 
-def _splitlines_no_ff(source):
-    """Split a string into lines ignoring form feed and other chars.
-
-    This mimics how the Python parser splits source code.
-    """
-    idx = 0
+_line_pattern = re.compile(r"(.*?(?:\r\n|\n|\r|$))")
+def _splitlines_no_ff(source, maxlines=-1):
     lines = []
-    next_line = ''
-    while idx < len(source):
-        c = source[idx]
-        next_line += c
-        idx += 1
-        # Keep \r\n together
-        if c == '\r' and idx < len(source) and source[idx] == '\n':
-            next_line += '\n'
-            idx += 1
-        if c in '\r\n':
-            lines.append(next_line)
-            next_line = ''
-
-    if next_line:
-        lines.append(next_line)
+    for lineno, match in enumerate(_line_pattern.finditer(source), 1):
+        if maxlines > 0 and lineno > maxlines:
+            break
+        lines.append(match[0].replace("\r\n", "\n"))
```

Comment:
The previous implementation tried to preserve the final `\r\n` together if I am not missing something. Why is this getting rid of `\r\n`s and converting them to `\n`s?

Reply:
Ah, that was my fault. I read the code wrong. I thought it was more complicated. Then the implementation would be even simpler.